tests/queries/0_stateless/test_merges_mutations_cancelling.sh

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env bash
+# Tags: replica, no-fasttest
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+
+. "$CURDIR"/../shell_config.sh
+
+# Drop existing tables if they exist.
+${CLICKHOUSE_CLIENT} --query="DROP TABLE IF EXISTS test_mutations_kill SYNC"
+
+# Test: Mutations Kill
+
+${CLICKHOUSE_CLIENT} --query="""
+CREATE TABLE test_mutations_kill
+(
+    c1 UInt64,
+    c2 String,
+    c3 UInt64,
+    c4 UInt64
+)
+ENGINE = ReplicatedMergeTree(
+    '/clickhouse/tables/$CLICKHOUSE_TEST_ZOOKEEPER_PREFIX/test_mutations_kill',
+    '1'
+)
+ORDER BY (c1, c2, c3);
+"""
+
+# Insert 4 million rows into test_mutations_kill.
+${CLICKHOUSE_CLIENT} --query="""
+INSERT INTO test_mutations_kill
+SELECT
+    number AS c1,
+    concat('string_', toString(number)) AS c2,
+    number % 100 AS c3,
+    number * 2 AS c4
+FROM numbers(4000000)
+SETTINGS min_insert_block_size_rows = 4000000, min_insert_block_size_bytes = 0;
+"""
+
+# Apply a mutation that deletes rows; the sleepEachRow function slightly delays processing.
+${CLICKHOUSE_CLIENT} --query="ALTER TABLE test_mutations_kill DELETE WHERE c1 > sleepEachRow(0.00001)"
+
+# Wait until the mutation is in progress.
+for i in {1..100}; do
+    # Check if the mutation merge started data reading (rows_read > 0), the cancelling didn't work at that phase before the fix in PR#77766
+    mutation_count=$(${CLICKHOUSE_CLIENT} --query="SELECT count() FROM system.merges WHERE table = 'test_mutations_kill' AND database = '$CLICKHOUSE_DATABASE' AND is_mutation AND rows_read > 0")
+
+    if [[ "$mutation_count" -gt 0 ]]; then
+        echo "Mutation started."
+        break
+    fi
+
+    if [[ "$i" -eq 100 ]]; then
+        echo "Mutation did not start in time."
+        exit 1
+    fi
+
+    sleep 0.1
+done
+
+# Note: The 'KILL MUTATION' can't be used in that test, because it only assigns a new 'fake' mutation which will make the killed one no-op, but it does not stop the mutation of parts already in progress.
+# ${CLICKHOUSE_CLIENT} --query="KILL MUTATION WHERE table = 'test_mutations_kill' AND database = '$CLICKHOUSE_DATABASE' FORMAT Vertical"
```

Comment:
BTW: @alesapin what do you think about that? Is it expected? Shouldn't KILL MUTATION actually kill the currectly running task (they can be long) and give it a chance to 'run' the 'killed' one instead?